custom_components/babybuddy/sensor.py

```diff
@@ -107,7 +107,9 @@ def services_changes_add(call):
             ATTR_TIME: call.data.get(ATTR_TIME),
             ATTR_WET: call.data.get(ATTR_WET),
             ATTR_SOLID: call.data.get(ATTR_SOLID),
-            ATTR_COLOR: call.data.get(ATTR_COLOR).lower(),
+            ATTR_COLOR: call.data[ATTR_COLOR].lower()
+            if ATTR_COLOR in call.data
+            else None,
```

Comment:
Have you had a chance to test this? I don't think "color" will be in call.data.get(ATTR_SOLID). Thinking it should be more like this?

    `ATTR_COLOR: call.data.get(ATTR_COLOR).lower() if isinstance(call.data.get(ATTR_COLOR), str) else call.data.get(ATTR_COLOR)`

Reply:
Yes it works, ˋ.get()ˋ return ˋNone ˋ if the key doesn’t exist